src/main/groovy/ca/stellardrift/gitpatcher/GitPatcher.groovy

```diff
@@ -67,6 +67,7 @@ class GitPatcher implements Plugin<Project> {
                 rootUpdate.configure { dependsOn(updateSubmodules) }
 
                 def apply = tasks.register('apply' + capitalizedName  +'Patches', ApplyPatchesTask) {
+                    setExtension(this.extension)
```

Comment:
For config cache purposes, we don't want to hold a reference to the extension instance itself -- just the properties that need to be tracked.

Reply:
How about just giving the ListProperty instance to it?